table/column.go

```diff
@@ -88,15 +88,22 @@ func ToColumn(col *model.ColumnInfo) *Column {
 // If pkIsHandle is false and name is ExtraHandleName, the extra handle column will be added.
 // If any columns don't match, return nil and the first missing column's name
 func FindCols(cols []*Column, names []string, pkIsHandle bool) ([]*Column, string) {
+	// Comparing strings more than (32 * 32 / 2) times cost a lot, So we use a map to do the finding
+	var colsMap map[string]*Column
+	if len(names) >= 32 {
+		colsMap = make(map[string]*Column)
```

Comment:
We can allocate the map with a hint, like
```go
colsMap = make(map[string]*Column, len(names))
```
So the hash map would allocate enough memory to avoid hash grow.

The minimum map has one bucket, a bucket can contain 8 elements, so we only need to set the hint when the size > 8.

Reply:
Addressed